### ISSUE TITLE
Document partially applying syntax using `_` 

### ISSUE THREAD

**Issue by chexxor** [purescript/documentation#52]
There isn't a single place in which all the `_` forms are documented, but there *is* docs for some specific syntaxes:
- [Operator sections](https://github.com/purescript/documentation/blob/master/language/Syntax.md#operator-sections)
- [Record updates](https://github.com/purescript/documentation/blob/master/language/Syntax.md#record-updates)

But I know it can also be done in `case..of` expressions, also, but that's not currently documented. Are there other places this can be used?

Should we document this as "Partial application of (special) syntactic expressions"?

**Comment by hdgarrood** [comment]
I'm not sure it makes sense to call all of these things "partial application"; there isn't really any partial application going in in `(case _ of x -> y)` or `(if _ then x else y)`. I'll check the source to get a complete list.

**Comment by garyb** [comment]
Those are the only others I can think of :smile: although you can put a wildcard in any or all of the `if` parts; `if _ then _ else _`.

**Comment by hdgarrood** [comment]
Yep; looking through `Language.PureScript.Sugar.ObjectWildcards` I think we've covered everything now. Not really relevant to this discussion but it seems like that module name no longer reflects what the code in the module does so accurately, since the wildcard syntax has been expanded? It's no longer just *object* wildcards.

**Comment by ysangkok** [comment]
Just stumbled upon the fact that

```
edit :: {y :: Int}
edit =
  let
    fun = _ { y = _ }
  in
    fun {y:5} 6
```

compiles but if you do `y = 1 + _` instead of `y = _` it doesn't. Syntax.md doesn't make it very clear that you can't do additional processing. After all, in operator sections you can. As a user I don't understand why I can't do the plus one.
